packages/shared/topology/sidebar/node/NodeDetails.tsx

```diff
@@ -0,0 +1,127 @@
+import * as React from 'react';
+import { getNodeRoles } from '@odf/core/utils';
+import { Timestamp } from '@odf/shared/details-page/timestamp';
+import { SectionHeading } from '@odf/shared/heading/page-heading';
+import { nodeStatus } from '@odf/shared/status/Node';
+import { Status } from '@odf/shared/status/Status';
+import { NodeKind } from '@odf/shared/types';
+import { useCustomTranslation } from '@odf/shared/useCustomTranslationHook';
+import { referenceForModel } from '@odf/shared/utils';
+import { ResourceLink } from '@openshift-console/dynamic-plugin-sdk';
+import * as _ from 'lodash-es';
+import { MachineModel } from './models';
+import NodeIPList from './NodeIPList';
+import { getNodeAddresses, getNodeMachineNameAndNamespace } from './utils';
+
+const cloudProviderID = (node) => {
+  return node.spec.providerID ? node.spec.providerID.split('://')[0] : '';
+};
```

Comment:
nits: not important, u can skip if u want...
```suggestion
const cloudProviderID = (node) => node?.spec?.providerID?.split('://')?.[0] || '';
```

Reply:
Done.